service.js

```diff
@@ -39,26 +41,31 @@ seneca.options.sanitizeTextArea = {
 };
 seneca.decorate('customValidatorLogFormatter', require('./lib/custom-validator-log-formatter'));
 seneca.use(store, config['postgresql-store']);
-seneca.use(require('./lib/cd-events'), {logger: log.logger});
+seneca.use(require('./lib/cd-events'), { logger: log.logger });
 seneca.use(require('cp-permissions-plugin'), {
   config: __dirname + '/config/permissions'
 });
 
 seneca.use(require('seneca-queue'));
 seneca.use(require('seneca-kue'));
-seneca.use(require('./lib/queues'), {config: config.kue});
+seneca.use(require('./lib/queues'), { config: config.kue });
+
 process.on('SIGINT', shutdown);
 process.on('SIGTERM', shutdown);
 process.on('uncaughtException', shutdown);
 process.on('SIGUSR2', shutdown);
 
 function shutdown (err) {
-  var stopQueue = seneca.export('queues/queue')['stopQueue'];
+  var stopQueue = seneca.export('queues/queue').stopQueue;
   stopQueue();
-  if (err !== void 0 && err.stack !== void 0) {
-    console.error(new Date().toString() + ' FATAL: UncaughtException, please report: ' + util.inspect(err));
-    console.error(util.inspect(err.stack));
-    console.trace();
+  if (err !== undefined && err.stack !== undefined) {
+    var error = {
+      date: new Date().toString(),
+      msg: 'FATAL: UncaughtException, please report: ' + util.inspect(err.stack),
+      err: util.inspect(err)
+    };
+    console.error(JSON.stringify(error));
+    process.exit(1);
```

Comment:
if the error has no stack, the error is swallowed ?

Reply:
thats how its always been. but fair point